main.nf

```diff
@@ -677,7 +677,9 @@ process MapReads {
         ${convertToFastq}
         bwa mem -K 100000000 -R \"${readGroup}\" ${extra} -t ${task.cpus} -M ${fasta} \
         ${input} | \
-        samtools sort --threads ${task.cpus} -m 2G - > ${idSample}_${idRun}.bam
+        samtools view -@ $task.cpus -b -h -O BAM -o ${idSample}_${idRun}.name.bam -
+
+        samtools sort -@ $task.cpus -o ${idSample}_${idRun}.bam -T ${idSample}_${idRun} ${idSample}_${idRun}.name.bam
```

Comment:
```suggestion
        samtools sort -@ $task.cpus -o ${idSample}_${idRun}.bam -T ${idSample}_${idRun} ${idSample}_${idRun}.name.bam
        rm -rf ${idSample}_${idRun}.name.bam
```

Reply:
Wouldn't this delete it even if samtools sort fails? Perhaps add a `&&` before?

Reply:
yes, but if samtools sort fails, then the process will fail, if that happen, we have automatic restart, and files waiting there to be deleted.
I think we should always delete them, no matter what...

Reply:
Fair enough

Reply:
I did think about removing that file but it's not great for debugging. `bwa mem` can take a few hours to run and if indeed there is a problem with the `sort` then at least we can check if `name.bam` has been generated properly. I would be tempted to leave it as it is for now. The `work` directory should be cleaned up afterwards anyway.

Reply:
By the way why `name.bam`? why not `unsorted.bam` ?

Reply:
The `samtools view` section converts the standard output of `bwa mem` (`.sam`) to BAM format so it's as storage friendly as possible.

Reply:
But true, for debugging, it'll be good to keep, but for a production pipeline, it's a good idea to remove it

Reply:
I believe `bwa` by default outputs a SAM/BAM that is actually sorted by the name of the reads because it maps the pairs together. You can check this by doing `samtools view file.bam | head`

Reply:
I think we should leave it as it is even in the production pipeline because the debugging is easier when users start using the pipeline. At that point, it would be good to be able to advise on possible checks as I mentioned earlier. We can try to be as efficient as possible with storage but ideally we apply that to the final results that are generated. If we need to generate and keep a file to make downstream debugging easier rather than saying "Um, I have no idea" then it makes sense.

Reply:
That does make sense, true, but then we will have to monitor space more closely

Reply:
But then, why not doing two different processes?

Reply:
That would be even better. Thats how I do it in all of my pipelines:
https://github.com/nf-core/chipseq/blob/7730c1f9cb707783bb5779ecdf4aee4e9e5a2f77/main.nf#L582-L614

Also, with the advent of `DSL2` will be easier to update because Im assuming we will need a separate module file to sort the BAM files.

Reply:
Why not in this PR as well then?

Reply:
Done :+1: